expression/builtin_string.go

```diff
@@ -3491,5 +3493,67 @@ type loadFileFunctionClass struct {
 }
 
 func (c *loadFileFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
-	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "load_file")
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+	valStr, _ := ctx.GetSessionVars().GetSystemVar(variable.MaxAllowedPacket)
+	maxAllowedPacket, err := strconv.ParseInt(valStr, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETString, types.ETString)
+	sig := &builtinInstrLoadFileSig{bf, maxAllowedPacket}
+	return sig, nil
+}
+
+type builtinInstrLoadFileSig struct {
+	baseBuiltinFunc
+	maxAllowedPacket int64
+}
+
+func (b *builtinInstrLoadFileSig) Clone() builtinFunc {
+	newSig := &builtinInstrLoadFileSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	newSig.maxAllowedPacket = b.maxAllowedPacket
+	return newSig
+}
+
+//MySQL implementation return NULL value when error occurs
+func (b *builtinInstrLoadFileSig) evalString(row chunk.Row) (string, bool, error) {
+	filePath, isNull, err := b.args[0].EvalString(b.ctx, row)
+	if isNull || err != nil {
+		return "", true, nil
+	}
+	//TODO: check filePath under secure_file_path(not exists in TiDB now)
```

Comment:
- - maybe we should ship this function after secure_file_path done?

it's too dangerous for someone can do `SELECT LOAD_FILE('/etc/passwd');`

Reply:
here besides the secure_file_path is needed as well as privlige control,   In MySQL SUPER FILE is needed to execute this function, this is another TODO @lysu

Reply:
It's a security vulnerability if we simply add this function.
An attacker can get any file he wants in on the host os.

Reply:
@cfzjywxk Could you file an issue about this TODO?